### PR TITLE
Fix getLocalAppDataPath for wsl

### LINKED ISSUE #74: WSL: Runtime error encountered
I get the following error in WSL when running smokehouse in lighthouse.

```
Runtime error encountered: { Error: EINVAL: invalid argument, mkdir '/mnt/c/Users/WardPeeters/Projects/os/lighthouse/node_modules/.bin:'
```

My path env is the following:
```
/mnt/c/Users/<user>/Projects/os/lighthouse/node_modules/.bin:/usr/local/share/.config/yarn/link/node_modules/.bin:/mnt/c/Users/<user>/Projects/os/lighthouse/node_modules/.bin:/usr/local/share/.config/yarn/link
/node_modules/.bin:/root/.nvm/versions/node/v8.9.3/lib/node_modules/npm/bin/node-gyp-bin:/root/.nvm/versions/node/v8.9.3/bin/node_modules/npm/bin/node-gyp-bin:/mnt/c/Users/<user>/Projects/os/lighthouse/node_modules/.b
in:/usr/local/share/.config/yarn/link/node_modules/.bin:/mnt/c/Users/<user>/Projects/os/lighthouse/node_modules/.bin:/usr/local/share/.config/yarn/link/node_modules/.bin:/root/.nvm/versions/node/v8.9.3/lib/node_module
s/npm/bin/node-gyp-bin:/root/.nvm/versions/node/v8.9.3/bin/node_modules/npm/bin/node-gyp-bin:/root/.nvm/versions/node/v8.9.3/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/mnt
/c/Program Files/Docker/Docker/resources/bin:/mnt/c/Windows/System32:/mnt/c/Windows:/mnt/c/Windows/System32/wbem:/mnt/c/Windows/System32/WindowsPowerShell/v1.0:/mnt/c/ProgramData/chocolatey/bin:/mnt/c/Program Files/dotnet:
/mnt/c/Program Files/Microsoft SQL Server/130/Tools/Binn:/mnt/c/Users/<user>/AppData/
```

### PR DESCRIPTION
**Pull request by wardpeet**
fixes #74

### CODE REVIEW

File: test/utils-test.ts [hunk 1]
```diff
@@ -30,4 +30,10 @@ describe('WSL path format to Windows', () => {
     const windows = 'D:\\Users\\user1\\AppData';
     assert.equal(toWinDirFormat(wsl), windows);
   });
+
+  it('getLocalAppDataPath returns a correct path', () => {
+    process.env.PATH = '/mnt/c/Users/user1/.bin:/mnt/c/Users/user1/AppData/';
+    const appDataPath = '/mnt/c/Users/user1/AppData/Local';
```
Comment (patrickhulce) [inline test/utils-test.ts]: ah, it was spanning the separator before? how about using `[^:]` then? 

I'm not sure if the original intention was to allow spanning multiple directories deep AppData

File: test/utils-test.ts [hunk 2]
```diff
@@ -30,4 +30,10 @@ describe('WSL path format to Windows', () => {
     const windows = 'D:\\Users\\user1\\AppData';
     assert.equal(toWinDirFormat(wsl), windows);
   });
+
+  it('getLocalAppDataPath returns a correct path', () => {
+    process.env.PATH = '/mnt/c/Users/user1/.bin:/mnt/c/Users/user1/AppData/';
```
Comment (patrickhulce) [inline test/utils-test.ts]: let's be sure to put path back to normal after this or better yet make this function accept it as input :)

File: src/utils.ts [hunk 3]
```diff
@@ -50,7 +50,7 @@ export function toWinDirFormat(dir: string = ''): string {
 
 export function getLocalAppDataPath(): string {
   const path = process.env.PATH;
-  const userRegExp = /\/mnt\/([a-z])\/Users\/(.+?)\/AppData\//;
+  const userRegExp = /\/mnt\/([a-z])\/Users\/([^\/]+)\/AppData\//;
```
Comment (patrickhulce) [inline src/utils.ts]: not sure if we need to be able to span a few directories deep 🤷‍♂️

Comment (wardpeet) [reply src/utils.ts]: as far as I know we don't, we just need to find the user and Drive letter.
/mnt/**c**/Users/**wardpeet**/AppData/

File: test/utils-test.ts (hunk 2 above)
Comment (wardpeet) [reply test/utils-test.ts]: yeah much better to have it as a param

File: test/utils-test.ts (hunk 1 above)
Comment (wardpeet) [reply test/utils-test.ts]: we don't need to span multiple directories deep so should I change this?

File: src/utils.ts (hunk 3 above)
Comment (patrickhulce) [reply src/utils.ts]: ah ok, sg then

let's go ahead and throw the colon in the regex too to make sure it doesn't span the separator either :)

File: test/utils-test.ts (hunk 1 above)
Comment (patrickhulce) [reply test/utils-test.ts]: discussed above, seems good to account for both :)

**Comment by patrickhulce** [comment]
@wardpeet looks like we've got some ts issue to work about but seems good to go after that! :)

**Comment by wardpeet** [comment]
@patrickhulce done ;)

**Comment by paulirish** [comment]
published in chrome-launcher@0.10.1

**Pull request merged**